### PR TITLE
Implement SourceLocation using Tasty reflect

### PR DESCRIPTION
**Pull request by nicolasstucki**
Testing in https://github.com/lampepfl/dotty-community-build/pull/30

### CODE REVIEW

File: shared/src/main/scala/minitest/api/SourceLocation.scala [hunk 1]
```diff
@@ -28,33 +27,18 @@ final case class SourceLocation(
...
+    val (fileName, path, line) = getSourceLocation
+    '(SourceLocation(Some(~fileName.toExpr), Some(~path.toExpr), ~line.toExpr))
+  }
 
-  //   def fromContext: Tree = {
-  //     val (fileNameExpr, pathExpr, lineExpr) = getSourceLocation
-  //     val SourceLocationSym = symbolOf[SourceLocation].companion
-  //     q"""$SourceLocationSym($fileNameExpr, $pathExpr, $lineExpr)"""
-  //   }
+  private def getSourceLocation(implicit tasty: Tasty): (String, String, Int) = {
```
Comment (allanrenucci) [inline shared/src/main/scala/minitest/api/SourceLocation.scala]: I suppose we can inline this in `fromContextImpl`. No need to maintain the keep the previous methods around

Comment (nicolasstucki) [reply shared/src/main/scala/minitest/api/SourceLocation.scala]: No we cannot. The top level splice requires a call to a static method.

**Comment by nicolasstucki** [comment]
This will fail with the latest changes. I will adapt it tomorrow on tonight's nightly build.

File: shared/src/main/scala/minitest/api/SourceLocation.scala (hunk 1 above)
Comment (allanrenucci) [reply shared/src/main/scala/minitest/api/SourceLocation.scala]: This is forbidden?
```scala
implicit transparent def fromContext: SourceLocation = ~fromContextImpl(TopLevelSplice.tastyContext)

def fromContextImpl(implicit tasty: Tasty): Expr[SourceLocation] = {
  val pos = tasty.rootPosition
  val file = pos.sourceFile
  val path = file.getParent.toString
  'SourceLocation(Some(~file.getFileName.toExpr), Some(~path.toExpr), ~pos.startLine.toExpr)
}
```

Comment (nicolasstucki) [reply shared/src/main/scala/minitest/api/SourceLocation.scala]: Oh, I missed. Yes, that one is possible.

**Pull request merged**